src/wrapper.rs

```
@@ -0,0 +1,242 @@
+
+      let result = q.execute(&mut *writer).await?;
+      Ok(WriteQueryResult {
+         rows_affected: result.rows_affected(),
+         last_insert_id: result.last_insert_rowid(),
+      })
+   }
+
+   /// Execute multiple write statements atomically within a transaction.
+   ///
+   /// This method:
+   /// 1. Begins a transaction (BEGIN)
+   /// 2. Executes all statements in order
+   /// 3. Commits on success (COMMIT)
+   /// 4. Rolls back on any error (ROLLBACK)
+   ///
+   /// The writer is held for the entire transaction, ensuring atomicity.
+   /// Returns the result of each statement execution.
+   pub async fn execute_transaction(
+      &self,
+      statements: Vec<(String, Vec<JsonValue>)>,
+   ) -> Result<Vec<WriteQueryResult>, Error> {
+      // Acquire writer for the entire transaction
+      let mut writer = self.inner.acquire_writer().await?;
+
+      // Begin transaction
+      sqlx::query("BEGIN").execute(&mut *writer).await?;
+
+      // Execute all statements, collecting results and rolling back on error
+      let result = async {
+         let mut results = Vec::new();
+         for (query, values) in statements {
+            let mut q = sqlx::query(&query);
+            for value in values {
+               q = bind_value(q, value);
+            }
+            let exec_result = q.execute(&mut *writer).await?;
+            results.push(WriteQueryResult {
+               rows_affected: exec_result.rows_affected(),
+               last_insert_id: exec_result.last_insert_rowid(),
+            });
+         }
+         Ok::<Vec<WriteQueryResult>, Error>(results)
+      }
+      .await;
+
+      // Commit or rollback based on result
+      match result {
+         Ok(results) => {
+            sqlx::query("COMMIT").execute(&mut *writer).await?;
+            Ok(results)
+         }
+         Err(e) => {
+            // Attempt rollback, but preserve original error
+            let _ = sqlx::query("ROLLBACK").execute(&mut *writer).await;
```

If rollback fails, the database could be left in an inconsistent state, and the user won't know.
Should we return a compound error that includes both original/rollback error?